datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/ECCurve.kt

```diff
@@ -15,87 +16,162 @@ import kotlinx.serialization.descriptors.SerialDescriptor
-                "0051 953EB961 8E1C9A1F 929A21A0 B68540EE A2DA725B 99B315F3 B8B48991 8EF109E1 56193951 EC7E937B 1652C0BD 3BB1BF07 3573DF88 3D2C34F1 EF451FD4 6B503F00"
-                    .replace(" ", "")
+                    "0051 953EB961 8E1C9A1F 929A21A0 B68540EE A2DA725B 99B315F3 B8B48991" +
+                "8EF109E1 56193951 EC7E937B 1652C0BD 3BB1BF07 3573DF88 3D2C34F1 EF451FD4" +
+                "6B503F00"
+        }.let {
+            coordinateCreator.parseString(string = it.replace(" ", ""), base = 16)
         }
-        modCreator.parseString(string = bString, base = 16)
     }
 
+    /**
+     * G: Generator of cyclic curve subgroup
+     * See https://www.secg.org/sec2-v2.pdf
+     */
+    val generator: CryptoPublicKey.EC by lazy {
+        when (this) {
+            SECP_256_R_1 ->
+                      "04 6B17D1F2 E12C4247 F8BCE6E5 63A440F2 77037D81 2DEB33A0" +
+                "F4A13945 D898C296 4FE342E2 FE1A7F9B 8EE7EB4A 7C0F9E16 2BCE3357" +
+                "6B315ECE CBB64068 37BF51F5"
+            SECP_384_R_1 ->
+                     "04 AA87CA22 BE8B0537 8EB1C71E F320AD74 6E1D3B62 8BA79B98" +
+               "59F741E0 82542A38 5502F25D BF55296C 3A545E38 72760AB7 3617DE4A" +
+               "96262C6F 5D9E98BF 9292DC29 F8F41DBD 289A147C E9DA3113 B5F0B8C0" +
+               "0A60B1CE 1D7E819D 7A431D7C 90EA0E5F"
+            SECP_521_R_1 ->
+                     "04 00C6858E 06B70404 E9CD9E3E CB662395 B4429C64 8139053F" +
+               "B521F828 AF606B4D 3DBAA14B 5E77EFE7 5928FE1D C127A2FF A8DE3348" +
+               "B3C1856A 429BF97E 7E31C2E5 BD660118 39296A78 9A3BC004 5C8A5FB4" +
+               "2C7D1BD9 98F54449 579B4468 17AFBD17 273E662C 97EE7299 5EF42640" +
+               "C550B901 3FAD0761 353C7086 A272C240 88BE9476 9FD16650"
+        }.replace(" ","").chunked(2).map { it.toInt(16).toByte() }.toByteArray()
+            .let(CryptoPublicKey.EC::fromAnsiX963Bytes)
+    }
+
+    /**
+     * n: Order of (the cyclic subgroup generated by) G
+     * See https://www.secg.org/sec2-v2.pdf
+     */
+    val order: BigInteger by lazy {
+        when(this) {
+            SECP_256_R_1 ->
+                "FFFFFFFF 00000000 FFFFFFFF FFFFFFFF BCE6FAAD A7179E84 F3B9CAC2" +
+                "FC632551"
+            SECP_384_R_1 ->
+                "FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF C7634D81" +
+                "F4372DDF 581A0DB2 48B0A77A ECEC196A CCC52973"
+            SECP_521_R_1 ->
+                "01FF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF" +
+                "FFFFFFFF FFFFFFFA 51868783 BF2F966B 7FCC0148 F709A5D0 3BB5C9B8" +
+                "899C47AE BB6FB71E 91386409"
+        }.replace (" ", "").toBigInteger(16)
+    }
+
+    /**
+     * h: Cofactor of the cyclic subgroup generated by G
+     * See https://www.secg.org/sec2-v2.pdf
+     */
+    val cofactor: Int get() =
+        when(this) {
+            SECP_256_R_1 -> 1
+            SECP_384_R_1 -> 1
+            SECP_521_R_1 -> 1
```

Comment:
those can be condensed into a single comma-separated case

Reply:
stylistically i'd prefer to keep it separate